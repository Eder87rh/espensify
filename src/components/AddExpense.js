import React from 'react'

export default () => (
    <div>
        This is from my add expense Component!!
    </div>
)