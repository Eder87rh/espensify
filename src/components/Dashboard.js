import React from 'react'

export default () => (
    <div>
        This is from my dashboard Component
    </div>
)