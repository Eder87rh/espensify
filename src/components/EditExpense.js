import React from 'react'

export default (props) => {
    console.log(props)
    return (
        <div>
            Editing the espense with id of { props.match.params.id }
        </div>
    )
}