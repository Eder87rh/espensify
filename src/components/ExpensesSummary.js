import React from 'react';
import { connect } from 'react-redux';
import { getTotalExpenser } from '../selectors/expenses-total'
import expensesFilter from '../selectors/expenses'
import numeral from 'numeral'

export const ExpensesSummary = ({ count, total }) => {
  console.log("TCL: ExpensesSummary -> total", total)
  const expenseWord = count === 1 ? 'expense' : 'expenses'; 
  const totalFormated = numeral(total /100).format('$0,0.00');

  return (
    <h1>
      Viewing { count } { expenseWord } totalling { totalFormated }
    </h1>
  );
}

const mapStateToProps = state => {
  const visibleExpenses = expensesFilter(state.expenses, state.filters)

  return {
    count: visibleExpenses.length,
    total: getTotalExpenser(expensesFilter(state.expenses, state.filters))
  }
}


 
export default connect(mapStateToProps)(ExpensesSummary);