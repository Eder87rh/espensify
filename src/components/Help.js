import React from 'react'

export default () => (<div>This is from my help expense PAGE!!</div>)