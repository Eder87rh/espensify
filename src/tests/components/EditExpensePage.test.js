import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpense';
import expenses from '../fixtures/expenses';

let editExpense, history, wrapper, removeExpense

beforeEach(() => {
  editExpense = jest.fn();
  removeExpense = jest.fn();
  history = { push: jest.fn() }

  wrapper = shallow(<EditExpensePage
    expense={expenses[0]}
    editExpense={editExpense}
    removeExpense={removeExpense}
    history={history}
  />)
})

describe('EditExpensePage tests', () => {
  test('EditExpensePage should render correctly without expenses', () => {
    const wrapper = shallow(<EditExpensePage />)
    expect(wrapper).toMatchSnapshot();
  })

  test('EditExpensePage should render correctly with expense data', () => {
    expect(wrapper).toMatchSnapshot();
  })
  
  test('EditExpensePage should save expense edited', () => {
    const editedExpense = {
      ...expenses[0],
      description: "Edited"
    }

    wrapper.find('ExpenseForm').prop('onSubmit')(editedExpense)
    expect(editExpense).toBeCalledWith(expenses[0].id, editedExpense)
    expect(history.push).toBeCalledWith("/")
  })

  test("EditExpensePage should be able to delete expenses", () => {
    wrapper.find("button").prop('onClick')(expenses[0])
    expect(removeExpense).toBeCalledWith({ id: expenses[0].id })
    expect(history.push).toBeCalledWith("/")
  })
});