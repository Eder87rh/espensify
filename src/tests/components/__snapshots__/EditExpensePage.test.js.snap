// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditExpensePage tests EditExpensePage should render correctly with expense data 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Gum",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;

exports[`EditExpensePage tests EditExpensePage should render correctly without expenses 1`] = `
<div>
  <ExpenseForm
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
